### ISSUE TITLE
Follow up on reclassified tests in #18899

### ISSUE THREAD

**Issue by odersky** [lampepfl/dotty#18904]
#18899 contains three test reclassifications:

 - i15922.scala was moved to pending. Not clear whether this should compile, and what changes would be necessary to get it there.
 - future-traverse.scala was moved to pending. Not clear how to make this compiler.
 - i15749a.scala was moved to neg. The issue description seems to indicate that the test should not compile, but I am not sure what the outcome should be.

We need to follow up on these and either classify them as definite errors / unimplementable or find alternative ways to make them compile.

**Comment by Linyxus** [comment]
I have reclassified these tests in [this commit](https://github.com/lampepfl/dotty/pull/18899/commits/032fe8aeaf7449cc0a26d8009ac1b3377d8a93ac).

Summary:

- For i15922, it should not work, as both bad1 and bad2 leak scoped
capabilities. It is moved to neg tests.
- For future-traverse.scala, we could make it work this way. Previously it won't
compile because we instantiated the type parameter of `successful` with a
universal capability, but actually we needn't do that: we could well pick a
local reach capability, as is done in this commit.
- For i15749a.scala, it should work but sadly not right now, due to a limitation
of our current implementation.
